test/demo/contextListener.test.ts

```diff
@@ -0,0 +1,46 @@
+import 'jest';
+import * as fdc3Remote from './utils/fdc3RemoteExecution';
+import {getFinConnection} from './utils/fin';
+
+
+describe('advanced puppeteer functionality', () => {
+    const testManagerIdentity = {uuid: 'test-app', name: 'test-app'};
+    const testIdentity1 = {uuid: 'test-app-1', name: 'test-app-1'};
+    const testIdentity2 = {uuid: 'test-app-2', name: 'test-app-2'};
+
+    it('can can register and trigger a context listener', async () => {
+        // Creating the apps can take a while, so we need a longer timeout.
+        jest.setTimeout(10000);
+
+        const fin = await getFinConnection();
+
+        // Check to see that the main window started correctly
+        expect(fin.Application.wrapSync(testManagerIdentity).isRunning()).resolves.toBe(true);
+
+        // Launch two test applications
+        await fdc3Remote.open(testManagerIdentity, testIdentity1.uuid);
+        await fdc3Remote.open(testManagerIdentity, testIdentity2.uuid);
+
+        // Check that the apps have started properly
+        expect(fin.Application.wrapSync(testIdentity1).isRunning()).resolves.toBe(true);
+        expect(fin.Application.wrapSync(testIdentity2).isRunning()).resolves.toBe(true);
+
+        // Register a listener. We don't await as the promise won't resolve until
+        // the listener is triggered
+        const callbackSpy = jest.fn();
+        const callbackResponse = fdc3Remote.addContextListener(testIdentity2, callbackSpy);
+        await new Promise(res => setTimeout(res,
```

Comment:
Feels inconsistent with `broadcast` that this `timeout` isn't going inside `fdc3Remote.addContextListener`. Noting the comment, though is there no way to resolve this inconsistency?

Reply:
I discussed with Phil at length yesterday. This is the only way that we can see to do it for now due to the weird workaround. We need to get the value back from the listener, which without "exposeFunction" can only be done by returning a nested promise which resolves when the listener is triggered. Since await nested promises awaits the whole stack, we can't (at least as far as I can see) know when the remote code has actually finished running. As far as adding the delay to the util, we would then lose the ability to get the value, unless i've missed some trick we could do with promises